src/disrupt.sh

```diff
@@ -1,43 +1,52 @@
 #!/bin/bash
+MO="${RUN_DIR}/lib/mo"
 
 RANDOM_FACT="$(shuf -n 1 "${FACTS_SOURCE}")"
+# This is exported so mo picks it up
+export RANDOM_FACT_HTML
 RANDOM_FACT_HTML="$(echo "${RANDOM_FACT}" | sed 's/&/\&amp;/g; s/</\&lt;/g; s/>/\&gt;/g; s/"/\&quot;/g; s/'"'"'/\&#39;/g')"
 
 DICE_ROLL=$(shuf -i 1-"${IMAGE_RARITY}" -n 1)
 
-MSG_TEMPLATE_IMG="${RUN_DIR}/assets/template_img"
-MSG_TEMPLATE_BASE="${RUN_DIR}/assets/template_head"
+TO_EMAIL_ADDRESS="$1"
+RENDERED_TEMPLATE=$(mktemp message.tmp.XXXXXXXXXXXXX)
 
-EMAIL_ADDRESS="$1"
-TEMP_FILE=$(mktemp message.tmp.XXXXXXXXXXXXX)
+TEMPLATE="${RUN_DIR}/assets/template.mo"
+PARTIAL="${RUN_DIR}/assets/partial"
 
-
-# add lucashifact
-cp "${MSG_TEMPLATE_BASE}" "${TEMP_FILE}"
-echo "${RANDOM_FACT_HTML}" >> "${TEMP_FILE}"
-echo '</p>' >> "${TEMP_FILE}"
-
-# if the directory doesn't exist
-# or no files inside the directory
-if [[ ! -d "${IMG_ASSET_DIR}" || ! $(find "${IMG_ASSET_DIR}" -mindepth 1 \( -iname '*.jpg' -o -iname '*.png' \) ) ]]; then
+# Set the dice roll to zero if:
+#   - the image assets directory doesn't exist
+#   - no supported image files exist inside the image assets directory
+if [[ ! -d "${IMG_ASSET_DIR}" || ! $(find "${IMG_ASSET_DIR}" -mindepth 1 -type f \( -iname '*.jpg' -o -iname '*.png' \) ) ]]; then
     DICE_ROLL=0
 fi
 
-# add the image if necessary
 if [ "${DICE_ROLL}" -eq 1 ]; then
-    RANDOM_IMAGE=$(find "${IMG_ASSET_DIR}" -type f | shuf -n 1)
-
-    cat "${MSG_TEMPLATE_IMG}" >> "${TEMP_FILE}"
-    base64 "${RANDOM_IMAGE}" >> "${TEMP_FILE}"
-else
-    # otherwise just end the body
-    echo '--body--' >> "${TEMP_FILE}"
+    # Setting INCLUDE_IMAGE allows us to conditionally include parts of the
+    # Mustache template
+    #
+    # We can't use the same DICE_ROLL variable since mo just checks whether
+    # the value is set to decide whether to include that part of the template
+    export INCLUDE_IMAGE=1
+
+    # Choose a random image from the image assets directory
+    RANDOM_IMAGE=$(find "${IMG_ASSET_DIR}" -mindepth 1 -type f \( -iname '*.jpg' -o -iname '*.png' \) | shuf -n 1)
```

Comment:
possible optimization - if we store the result of this find above we can just use that as input to shuf, right?

Reply:
:+1: